packages/format-library/src/link/inline.js

```diff
@@ -127,7 +167,11 @@ function InlineLinkUI( {
 			onClose={ stopAddingLink }
 			position="bottom center"
 		>
-			<LinkControl value={ linkValue } onChange={ onChangeLink } />
+			<LinkControl
+				value={ linkValue }
+				onChange={ onChangeLink }
+				forceIsEditingLink={ addingLink }
```

Comment:
I'd have preferred to avoid knowing that a "mode" exists outside the component but I don't have an alternative.

Reply:
> I'd have preferred to avoid knowing that a "mode" exists outside the component but I don't have an alternative.

I am on the fence about it. I think based on discussion in #18061, the statefulness of toggling between previewing and editing is a key feature of the component. I agree that this is (and should be) normally handled internal to the component itself, but I don't think it needs to be hidden so far as being a publicly-acknowledged feature. 

Allowing these to be controlled is an awkward position, and I weighed whether this should justify some rethink about whether it actually be managed as internal state vs. provided by the rendering parent. I think the conveniences and primary use-cases of the component should still defend that it continue to exist as state of the component.

The use of `force`-prefixed state controlling props does have some prior art as well, notably in the `PostPreviewButton` component ([`forcePreviewLink`, `forceIsAutosaveable`](https://github.com/WordPress/gutenberg/blob/38f4914149bc6fecbeb45e86ed29b2c101ca65fc/packages/editor/src/components/post-preview-button/index.js#L205)).